### PR TITLE
Gdr 2371 2

### PR DESCRIPTION
**Pull request by gladkia**
# Description
Add GHA pipeline template for pkgdown.
It will be used by all gdrplatform packages on GitHub.
Preview: https://gdrplatform.github.io/gDRstyle

More details can be found at: https://docs.google.com/document/d/1WGW4_lOOZ73lEfVZwIWMivbyCgDTJAajxP5N8767DZY.

## What changed?
Related JIRA issue:  GDR-2371

## Why was it changed?
Enable static webpages with documentation for all packages on GitHub. 

# Logistic checklist
- [x] Package version bumped
- [x] Changelog updated

### CODE REVIEW

File: .github/workflows/pkgdown.yaml [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+      USERNAME:
+        required: true
+        type: string
+      PACKAGE_NAME:
+        required: true
+        type: string
+      BRANCH_NAME:
+        required: true
+        type: string
+      DEFAULT_BRANCH:
```
Comment (gladkia) [inline .github/workflows/pkgdown.yaml:23]: @marcin-kam I would remove it. We define supported branches in l9. What do you think?

File: .github/workflows/pkgdown.yaml [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    # Only restrict concurrency for non-PR jobs
+    concurrency:
+      group: pkgdown-${{ github.event_name != 'pull_request' || github.run_id }}
+    env:
+      GITHUB_PAT: ${{ secrets.GITHUB_TOKEN }}
+      TEST_TAG: user/app:latest-${{ github.sha }}
+      USERNAME: ${{ github.actor }}
+      PACKAGE_NAME: ${{ github.event.repository.name }}
+      BRANCH_NAME: ${{ github.head_ref }}
+      BASE_IMAGE: marcinkam/gdrshiny:1.0
```
Comment (gladkia) [inline .github/workflows/pkgdown.yaml:45]: @marcin-kam - the image you suggested works great.

File: .github/workflows/pkgdown.yaml [hunk 3]
```diff
@@ -0,0 +1,72 @@
+# This workflow is used to update static webpage with documentation for the given GitHub repository.
+# It's triggered by the push of the default branch. 
+# More details: https://docs.google.com/document/d/1WGW4_lOOZ73lEfVZwIWMivbyCgDTJAajxP5N8767DZY
+
+name: pkgdown
+
+on:
+  push:
+    branches: [main, master, test_ci_pkgdown]
```
[the remark this answers is not in the document]
Comment (gladkia) [reply .github/workflows/pkgdown.yaml:9]: Do you mean 'master' as a default branch for all packages? It sounds great to me.

File: .github/workflows/pkgdown.yaml (hunk 1 above)
Comment (marcin-kam) [reply .github/workflows/pkgdown.yaml:23]: Sure, I guess it's not needed in this workflow

**Review by marcin-kam — APPROVED** [review]

File: .github/workflows/pkgdown.yaml (hunk 1 above)
Comment (gladkia) [reply .github/workflows/pkgdown.yaml]: done.

**Pull request merged**